server/services/interventionsService.test.ts

```diff
@@ -2846,6 +2846,59 @@ pactWith({ consumer: 'Interventions UI', provider: 'Interventions Service' }, pr
     })
   })
   */
+
+  describe('getServiceProviderReportingData', () => {
+    const reportingResponse = [
+      {
+        referralLink: 'https://refer-and-monitor.example/referral/c9f9e22f-ddd9-422a-a9af-76fc02b18b38',
+        referralRef: 'SM1973AC',
+        referralId: 'c9f9e22f-ddd9-422a-a9af-76fc02b18b38',
+        contractId: 'thing',
+        organisationId: 'XYZ5678',
+        caseworkerId: 'liane.supplier@example.com',
+        serviceUserCrn: 'X017844',
+        dateReferralReceived: '2021-06-27T14:49:01+01:00',
+        dateSaaBooked: '2021-06-27T14:49:01+01:00',
+        dateSaaAttended: '2021-06-27T14:49:01+01:00',
+        dateFirstActionPlanSubmitted: '2021-06-27T14:49:01+01:00',
+        dateOfFirstActionPlanApproval: '2021-06-27T14:49:01+01:00',
+        dateOfFirstAttendedSession: '2021-06-27T14:49:01+01:00',
+        outcomesToBeAchievedCount: 8,
+        outcomesAchieved: 5.5,
+        countOfSessionsExpected: 10,
+        countOfSessionsAttended: 6,
+        endRequestedByPPAt: '2021-06-27T14:49:01+01:00',
+        endRequestedByPPReason: 'REC',
+        dateEOSRSubmitted: '2021-06-27T14:49:01+01:00',
+        concludedAt: '2021-06-27T14:49:01+01:00',
+      },
+    ]
+
+    it('returns a list of referrals for with data for reporting', async () => {
+      await provider.addInteraction({
+        state: 'nothing',
+        uponReceiving: 'a GET request for referral reporting data',
+        withRequest: {
+          method: 'GET',
+          path: '/performance-report',
+          query: { fromIncludingDate: Matchers.like('2021-06-01'), toIncludingDate: Matchers.like('2021-06-10') },
```

Comment:
Why do we need the `Matchers` here? Aren't these the exact values that we expect to be sent in the query, given the parameters passed to `getServiceProviderReportingData` below?

Reply:
I think I thought this was relevant for the Provider tests (ie. so they could put any value in on their side and not be tied to the dates we've put in here) - I think I'm misunderstanding that, right? Is this more useful for us for things like values that might depend on a random value like a uuid that we don't really care about?

Reply:
I've updated this

Reply:
I think that we usually use the matchers when making assertions about things that we're not fully in control of - e.g. some ID in a response, a timestamp that depends on the moment the request is made, etc. We use them to say "we expect to get some data that looks like this". But here, we are fully in control of how this request is constructed - the `fromIncludingDate` and `toIncludingDate` are just what we pass to the `getServiceProviderReportingData` method in the test.